packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -465,11 +480,19 @@ export class RunningSummarizer implements IDisposable {
         this.summarizing = new Deferred<void>();
 
         (async () => {
-            const result = await this.summarize(reason, false);
-            if (result !== true) {
-                // On nack or error, try again in safe mode
-                await this.summarize(reason, true);
+            if (true === await this.summarize(reason, { refreshLatestAck: false, fullTree: false })) {
+                return;
+            }
+            // On nack or error, try again fetching latest from storage server
+            if (true === await this.summarize(reason, { refreshLatestAck: true, fullTree: false })) {
+                return;
+            }
+            // On another failure, run the full tree
+            if (true === await this.summarize(reason, { refreshLatestAck: true, fullTree: true })) {
+                return;
             }
+            // If all 3 attempts failed, close the summarizer container
+            this.internalsProvider.stop("failToSummarize");
```

Comment:
Should there be an error event written here? I.e. it should be rare and almost every such case feels like should be a bug, so we should immediately look into it, vs. reporting reason for summarizer closure being general event that we will look mostly in aggregate.

Alternatively, we can make eventName: "StoppingSummarizer" use different categories (including "error" category) for some of the reasons (for now - only this reason). It's not great that events would land in different tables, but we are already in the same boat RE some _cancel events, so I always use unions :)

Reply:
This is a good point.. I'll try to upgrade this to an error event.